solr/core/src/java/org/apache/solr/cloud/RecoveryStrategy.java

```diff
@@ -220,7 +220,7 @@ protected String getReplicateLeaderUrl(ZkNodeProps leaderprops) {
   final private void replicate(String nodeName, SolrCore core, ZkNodeProps leaderprops)
       throws SolrServerException, IOException {
 
-    final String leaderUrl = getReplicateLeaderUrl(leaderprops);
+    final String leaderUrl = getReplicateLeaderUrl(leaderprops);//TODO?
```

Comment:
Did you say TODO because maybe this URL needs to be checked here?  I did a "Analyze Data Flow To Here" in IntelliJ at the spot where we fetch the URL.  This URL is always sourced from the SolrCloud cluster state, and not a parameter.  So we're good.  That said, it could be worth a comment in some places to say that we "trust" the leaderUrl from the cluster state (needn't be checked)

Reply:
Exactly. Thanks for checking!